cpp/include/compute_partition.cuh

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <thrust/binary_search.h>
+
+#include <experimental/graph.hpp>
+
+#include <rmm/thrust_rmm_allocator.h>
+
+namespace cugraph {
+namespace experimental {
+namespace detail {
+
+template <typename graph_view_type>
+class compute_partition_t {
+ public:
+  using graph_view_t = graph_view_type;
+  using vertex_t     = typename graph_view_type::vertex_type;
+
+  compute_partition_t(graph_view_t const &graph_view)
+  {
+    init<graph_view_t::is_multi_gpu>(graph_view);
+  }
+
+  template <bool is_multi_gpu, typename std::enable_if_t<!is_multi_gpu> * = nullptr>
+  void init(graph_view_t const &graph_view)
+  {
+  }
+
+  template <bool is_multi_gpu, typename std::enable_if_t<is_multi_gpu> * = nullptr>
+  void init(graph_view_t const &graph_view)
+  {
+    auto partition = graph_view.get_partition();
+    row_size_      = partition.get_row_size();
+    col_size_      = partition.get_col_size();
```

Comment:
Better take handle, and get row_comm_size_, col_comm_size_, and comm_size_ from handle.

row_comm_size_  = handle.get_subcomm(cugraph::partition_2d::key_naming_t().row_name()).get_size();